### PR TITLE
docs: adding introduction page

### PR DESCRIPTION
**Pull request by dbanksdesign**
#### Description of changes

Adding the 'Introduction' page to explain our vision for Amplify UI. This page is meant to be framework-agnostic. 

Also fixed the CRA and NextJS pages to not show up on non-react frameworks. 

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are updated
- [x] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated
- [x] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/src/pages/[platform]/getting-started/introduction/index.page.mdx [hunk 1]
```diff
@@ -0,0 +1,64 @@
+---
+title: Introduction
+description: What is Amplify UI?
+supportedFrameworks: all
+---
+
+import { Alert } from '@aws-amplify/ui-react';
+
+Amplify UI is an open-source UI library that brings the simplicity and extensibility of AWS Amplify to UI development. It consists of connected components that simplify complex workflows like authentication and dynamic data, primitive components that form the building blocks of a UI, and themes to make Amplify UI to fit any brand. **Extensibility and customization** are at the forefront of Amplify UI allowing easy to integration into any application regardless of the front-end tech stack.
```
Comment (hbuchel) [inline docs/src/pages/[platform]/getting-started/introduction/index.page.mdx:9]: Extra to?

```suggestion
Amplify UI is an open-source UI library that brings the simplicity and extensibility of AWS Amplify to UI development. It consists of connected components that simplify complex workflows like authentication and dynamic data, primitive components that form the building blocks of a UI, and themes to make Amplify UI fit any brand. **Extensibility and customization** are at the forefront of Amplify UI allowing easy to integration into any application regardless of the front-end tech stack. 
```

File: docs/src/pages/[platform]/getting-started/introduction/index.page.mdx [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+## Primitive components
+
+<Alert variation="info">
+  Primitive components are React-only for now
+</Alert>
+
+Amplify UI primitive components are in the middle of the spectrum between 'headless' UI frameworks like [Radix](https://www.radix-ui.com/) and [Headless UI](https://headlessui.dev/), and 'batteries included' frameworks like [MUI](https://mui.com/) and [Chakra](https://chakra-ui.com/). We want to provide a solid base that doesn't force a particular front-end architecture and every detail can be customized.
+
+All the styling is handled by plain CSS and theme-able with CSS variables. However we do provide some extra styling utilities in React like responsive style props, ThemeProvider and useTheme hook, useBreakpointValue for responsive values.
```
Comment (hbuchel) [inline docs/src/pages/[platform]/getting-started/introduction/index.page.mdx:27]: Missing and

```suggestion
All the styling is handled by plain CSS and theme-able with CSS variables. However we do provide some extra styling utilities in React like responsive style props, ThemeProvider and useTheme hook, and useBreakpointValue for responsive values.
```

**Review by hbuchel — APPROVED** [review]
Suggested some minor typo edits, otherwise LGTM!

File: docs/src/pages/[platform]/getting-started/introduction/index.page.mdx [hunk 3]
```diff
@@ -0,0 +1,64 @@
+---
+title: Introduction
+description: What is Amplify UI?
+supportedFrameworks: all
+---
+
+import { Alert } from '@aws-amplify/ui-react';
+
+Amplify UI is an open-source UI library that brings the simplicity and extensibility of AWS Amplify to UI development. It consists of connected components that simplify complex workflows like authentication and dynamic data, primitive components that form the building blocks of a UI, and themes to make Amplify UI fit any brand. **Extensibility and customization** are at the forefront of Amplify UI allowing easy to integration into any application regardless of the front-end tech stack.
```
Comment (joebuono) [inline docs/src/pages/[platform]/getting-started/introduction/index.page.mdx:9]: nit typo: "allowing easy ~~to~~ integration"

File: docs/src/pages/[platform]/getting-started/introduction/index.page.mdx [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+## Primitive components
+
+<Alert variation="info">
+  Primitive components are React-only for now
+</Alert>
+
+Amplify UI primitive components are in the middle of the spectrum between 'headless' UI frameworks like [Radix](https://www.radix-ui.com/) and [Headless UI](https://headlessui.dev/), and 'batteries included' frameworks like [MUI](https://mui.com/) and [Chakra](https://chakra-ui.com/). We want to provide a solid base that doesn't force a particular front-end architecture and every detail can be customized.
+
+All the styling is handled by plain CSS and theme-able with CSS variables. However we do provide some extra styling utilities in React like responsive style props, ThemeProvider and useTheme hook, and useBreakpointValue for responsive values.
```
Comment (joebuono) [inline docs/src/pages/[platform]/getting-started/introduction/index.page.mdx:27]: Perhaps we should have internal links to some of these styling utilities?
- [ThemeProvider](https://ui.docs.amplify.aws/react/theming/dark-mode#themeprovider)
- [useThemeHook](https://ui.docs.amplify.aws/react/theming#breakpoints:~:text=//%20Option%202%3A%20Get%20the%20theme%20object%20through%20the%20useTheme%20hook%20and%20style%20components%20with%20it)
- [useBreakPointValue](https://ui.docs.amplify.aws/react/theming/responsive#usebreakpointvalue)

File: docs/src/pages/[platform]/getting-started/introduction/index.page.mdx [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+Good components are fast and use native styling systems when possible, adding less dependencies, smaller memory footprint, and minimizing side effects.
+
+**Escape Hatches over ejections**
+Users should be able to control any aspect of our components using their favorite tools without throwing everything away. We meet users where they are and play nicely with platforms and frameworks.
+
+**Accessible by default**
+Component implementations follows platform accessibility standards and best practices (e.g. WCAG 2.1AA for Web, Apple HIG for iOS).
+
+**Respect the platform**
+We want to share as much as possible between platforms (like themes and schemas), and respect each platform identity and rules when possible (e.g. following name conventions and idioms). React components should feel like React components.
```
Comment (reesscot) [inline docs/src/pages/[platform]/getting-started/introduction/index.page.mdx:43]:
```suggestion
We want to share as much as possible between platforms (like themes and schemas), and respect each platform identity and rules when possible (e.g. following name conventions and idioms). For example, React components should feel like React components.
```

**Review by reesscot — APPROVED** [review]

**Pull request merged**